src/components/Reown/modals/SendTransactionModal.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect } from 'react';
+import { t } from 'ttag';
+import { useSelector, useDispatch } from 'react-redux';
+import { constants, numberUtils } from '@hathor/wallet-lib';
+import { unregisteredTokensDownloadRequested } from '../../../actions';
+import { CopyButton } from '../../CopyButton';
+
+export function SendTransactionModal({ data, firstAddress, onAccept, onReject }) {
+  const dispatch = useDispatch();
+  const { tokenMetadata, tokens: registeredTokens } = useSelector((state) => ({
+    tokenMetadata: state.tokenMetadata,
+    tokens: state.tokens,
+  }));
+
+  useEffect(() => {
+    // Collect all unregistered token UIDs
+    const unregisteredTokens = new Set();
+    
+    data?.data?.inputs?.forEach(input => {
+      const token = registeredTokens.find(t => t.uid === input.token);
+      if (!token && input.token) {
+        unregisteredTokens.add(input.token);
+      }
+    });
+
+    data?.data?.outputs?.forEach(output => {
+      const token = registeredTokens.find(t => t.uid === output.token);
+      if (!token && output.token) {
+        unregisteredTokens.add(output.token);
+      }
+    });
+
+    if (unregisteredTokens.size > 0) {
+      dispatch(unregisteredTokensDownloadRequested(Array.from(unregisteredTokens)));
+    }
+  }, [data, registeredTokens, dispatch]);
+
+  const getTokenSymbol = (tokenId) => {
+    if (!tokenId) {
+      return constants.DEFAULT_NATIVE_TOKEN_CONFIG.symbol;
+    }
+
+    const token = registeredTokens.find(t => t.uid === tokenId);
+    if (token) {
+      return token.symbol;
+    }
+
+    return '?';
+  };
+
+  const formatValue = (value) => {
+    if (value == null) {
+      return '-';
+    }
+
+    return numberUtils.prettyValue(value);
```

Comment:
We should consider if the token is an NFT here as well.

Reply:
Done, thanks